lib/em/protocols/linetext2.rb

```diff
@@ -57,6 +57,7 @@ def receive_data data
         while remaining_data.length > 0
           if @lt2_mode == :lines
             if ix = remaining_data.index( @lt2_delimiter )
+              delimiter_length = remaining_data.slice(@lt2_delimiter).length
```

Comment:
My markdown was poor...here is the code segment:

`if delimiter_string = remaining_data.slice( @lt2_delimiter )`
`  ix = remaining_data.index(delimiter_string)`
`  delimiter_length = delimiter_string.length`

Actually, we could just get rid of the `delimiter_length` variable and put `delimiter_string.length` in the appropriate place lower down.  I'll do that and update.